components/org.wso2.carbon.identity.oauth.dcr/src/main/java/org/wso2/carbon/identity/oauth/dcr/util/DCRConfigUtils.java

```diff
@@ -0,0 +1,267 @@
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.carbon.identity.oauth.dcr.util;
+
+import org.apache.commons.collections.CollectionUtils;
+import org.apache.commons.lang.StringUtils;
+import org.wso2.carbon.identity.configuration.mgt.core.ConfigurationManager;
+import org.wso2.carbon.identity.configuration.mgt.core.exception.ConfigurationManagementException;
+import org.wso2.carbon.identity.configuration.mgt.core.model.Attribute;
+import org.wso2.carbon.identity.configuration.mgt.core.model.Resource;
+import org.wso2.carbon.identity.configuration.mgt.core.model.ResourceAdd;
+import org.wso2.carbon.identity.core.util.IdentityUtil;
+import org.wso2.carbon.identity.oauth.common.OAuthConstants;
+import org.wso2.carbon.identity.oauth.dcr.DCRMConstants;
+import org.wso2.carbon.identity.oauth.dcr.exception.DCRMClientException;
+import org.wso2.carbon.identity.oauth.dcr.exception.DCRMServerException;
+import org.wso2.carbon.identity.oauth.dcr.internal.DCRDataHolder;
+import org.wso2.carbon.identity.oauth.dcr.model.DCRConfiguration;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+import static org.wso2.carbon.identity.configuration.mgt.core.constant.ConfigurationConstants.ErrorMessages.ERROR_CODE_RESOURCE_DOES_NOT_EXISTS;
+import static org.wso2.carbon.identity.configuration.mgt.core.constant.ConfigurationConstants.ErrorMessages.ERROR_CODE_RESOURCE_TYPE_DOES_NOT_EXISTS;
+import static org.wso2.carbon.identity.oauth.dcr.DCRMConstants.CLIENT_AUTHENTICATION_REQUIRED;
+import static org.wso2.carbon.identity.oauth.dcr.DCRMConstants.DCRConfigErrorMessage.ERROR_CODE_DCR_CONFIGURATION_RETRIEVE;
+import static org.wso2.carbon.identity.oauth.dcr.DCRMConstants.DCR_CONFIG_RESOURCE_NAME;
+import static org.wso2.carbon.identity.oauth.dcr.DCRMConstants.DCR_CONFIG_RESOURCE_TYPE_NAME;
+import static org.wso2.carbon.identity.oauth.dcr.DCRMConstants.ENABLE_FAPI_ENFORCEMENT;
+import static org.wso2.carbon.identity.oauth.dcr.DCRMConstants.MANDATE_SSA;
+import static org.wso2.carbon.identity.oauth.dcr.DCRMConstants.SSA_JWKS;
+import static org.wso2.carbon.identity.oauth.dcr.util.DCRConfigErrorUtils.handleClientException;
+import static org.wso2.carbon.identity.oauth.dcr.util.DCRConfigErrorUtils.handleServerException;
+
+
+/**
+ * Util class for DCR configurations and DCR resource related operations.
+ */
+public class DCRConfigUtils {
+
+    private DCRConfigUtils() { }
+
+    /**
+     * Persist the DCRConfiguration object.
+     *
+     * @param dcrConfiguration The DCRConfiguration object.
+     */
+    public static void setDCRConfiguration(DCRConfiguration dcrConfiguration)
```

Comment:
use util class for the generic logic only

Reply:
The `getDCRConfiguration()` method inside `DCRConfigUtils` is static. And it is called inside both `DCRConfigurationMgtServiceImpl` and `DCRMService` classes.

If we move the `getDCRConfiguration()` logic to `DCRConfigurationMgtServiceImpl` we cannot have the static behaviour since `DCRConfigurationMgtServiceImpl` is picked from osgi in the `identity-api-server` level and requires a single object to be created and kept [1] similar to [2]. And we need to have the static behaviour in `DCRMService`, since we don't want to create new `DCRConfigurationMgtServiceImpl` instances each time the DCR endpoint is hit.
Therefore it seems fine that we keep the logic in `DCRConfigUtils` class. Since this util class is DCR config specific it won't get any big later.

[1] DCRConfigurationMgtService getInstance()](https://github.com/anjuchamantha/identity-api-server/blob/36ea2f1de017f399634e017d11ecc9f7effd4ed1/components/org.wso2.carbon.identity.api.server.configs/org.wso2.carbon.identity.api.server.configs.common/src/main/java/org/wso2/carbon/identity/api/server/configs/common/factory/DCRMgtOGSiServiceFactory.java#L43) 
[2] JWTClientAuthenticatorMgtService getInstance()](https://github.com/wso2/identity-api-server/blob/1ebe8c5568926d8a453160fb571d5d440d5eb2d9/components/org.wso2.carbon.identity.api.server.configs/org.wso2.carbon.identity.api.server.configs.common/src/main/java/org/wso2/carbon/identity/api/server/configs/common/factory/JWTAuthenticationMgtOGSiServiceFactory.java#L43)

@Thumimku @shashimalcse please check

Reply:
Disregard the previous comment. Fixed with https://github.com/wso2-extensions/identity-inbound-auth-oauth/pull/2344/commits/f46ffbf511c8d774f6fff8be5b93b788f9515358